### PR TITLE
GH-15017: [Python] Harden test_memory.py for use with ARROW_USE_GLOG=ON

### LINKED ISSUE #15017: [Python] BUG: two failing tests related to ARROW_DEFAULT_MEMORY_POOL in conda builds on unix
### Describe the bug, including details regarding any error messages, version, and platform.

#14832 switched on the test suite within the conda-builds, which itself synced with https://github.com/conda-forge/arrow-cpp-feedstock/pull/875

In the process of doing two tests failing with the same error were skipped,
https://github.com/apache/arrow/blob/2c768a1767c0d7b33644910e0e96a949bc6f6f42/dev/tasks/conda-recipes/arrow-cpp/meta.yaml#L380-L381

That error looks like:
```
FAILED test_memory.py::test_env_var - assert 2 == 1
 +  where 2 = len(['WARNING: Logging before InitGoogleLogging() is written to STDERR', "W20221123 12:16:17.825700 443543 memory_pool.cc:113] Unsupported backend 'nonexistent' specified in ARROW_DEFAULT_MEMORY_POOL (supported backends are 'mimalloc', 'jemalloc', 'system')"])
```
which might indicate a missing default somewhere?

To fix this, the line quoted above should be removed, and the conda tests still pass.

### Component(s)

C++, Continuous Integration, Python

### PR DESCRIPTION
**Pull request by h-vetinari**
Accept output pattern for ARROW_USE_GLOG=ON too.

* Closes: #15017

### CODE REVIEW

**Comment by h-vetinari** [comment]
@github-actions crossbow submit -g conda-linux-x64-cpu-py3

**Comment by h-vetinari** [comment]
I don't want to run the full barrage of conda-tests here to verify that the tests passes when glog is enabled. One run of `conda-{linux,osx}-x64-cpu-py3` would be enough, but I don't know how to do that with the crossbow invocation.

**Comment by kou** [comment]
@github-actions crossbow submit conda-linux-x64-cpu-py3 conda-osx-x64-cpu-py3

**Comment by h-vetinari** [comment]
@github-actions crossbow submit conda-linux-x64-cpu-py3 conda-osx-x64-cpu-py3

File: python/pyarrow/tests/test_memory.py [hunk 1]
```diff
@@ -134,8 +134,14 @@ def check_env_var(name, expected, *, expect_warning=False):
...
     if expect_warning:
-        assert len(errlines) == 1
-        assert f"Unsupported backend '{name}'" in errlines[0]
+        assert len(errlines) in (1, 2)
+        if len(errlines) == 1:
+            # ARROW_USE_GLOG=OFF
+            assert f"Unsupported backend '{name}'" in errlines[0]
+        else:
+            # ARROW_USE_GLOG=ON
+            assert f"InitGoogleLogging()" in errlines[0]
```
Comment (kou) [inline python/pyarrow/tests/test_memory.py:143]:
```suggestion
            assert "InitGoogleLogging()" in errlines[0]
```

**Review by kou — APPROVED** [review]
+1

**Comment by h-vetinari** [comment]
Thanks for fixing the superfluous f-string! :)

**Pull request merged**